client/src/pages/Configurations/index.test.tsx

```diff
@@ -10,24 +10,124 @@ const renderPageView = () =>
     </BrowserRouter>
   );
 
-describe('Configurations', () => {
-  it('should contain a table with certain columns', async () => {
-    renderPageView();
-
-    expect(screen.getByTestId('table')).toBeInTheDocument();
-  });
-  it('should contain a call-to-action button', async () => {
+describe('Configurations Page', () => {
+  // Test case 1: Rendering the Configurations page
+  test('renders the Configurations page with title and search bar', () => {
     renderPageView();
     expect(
-      screen.getByText('Set up new condition', {
-        selector: 'button.usa-button[type=button]',
-      })
+      screen.getByText('Your reportable condition configurations')
     ).toBeInTheDocument();
-  });
-  it('should have a search component with correct placeholder', async () => {
-    renderPageView();
     expect(
       screen.getByPlaceholderText('Search configurations')
     ).toBeInTheDocument();
+    expect(
+      screen.getByRole('button', { name: 'Set up new condition' })
+    ).toBeInTheDocument();
+  });
+
+  // Test case 2: 'Set up new condition' button click opens the modal
+  test('opens the modal when "Set up new condition" button is clicked', () => {
+    renderPageView();
+    const setUpButton = screen.getByRole('button', {
+      name: 'Set up new condition',
+    });
+    fireEvent.click(setUpButton);
+    expect(screen.getByRole('dialog')).toBeInTheDocument();
+    expect(screen.getByText('Set up new condition')).toBeInTheDocument();
+  });
+
+  // Test case 3: Selecting a condition from the ComboBox
+  test('selects a condition from the ComboBox and enables the Add condition button', async () => {
+    renderPageView();
+    const setUpButton = screen.getByRole('button', {
+      name: 'Set up new condition',
+    });
+    fireEvent.click(setUpButton);
+
+    // Expect the modal to open
+    await waitFor(() => {
+      expect(screen.queryByRole('dialog')).toBeInTheDocument();
+    });
```

Comment:
I can't recall if this could be related to the error you mentioned you were seeing but I believe the preferred way to write this is:
```suggestion
    // Expect the modal to open
    const modal = await screen.findByRole('dialog');
    expect(modal).toBeInTheDocument();
```
The `findBy` functions automatically perform the waiting for you.